internal/handlers/sqlite/msg_getlog.go

```diff
@@ -16,12 +16,137 @@ package sqlite
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"time"
 
+	"go.uber.org/zap"
+
+	"github.com/FerretDB/FerretDB/build/version"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonparams"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/logging"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 // MsgGetLog implements HandlerInterface.
 func (h *Handler) MsgGetLog(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
-	return nil, notImplemented(must.NotFail(msg.Document()).Command())
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	command := document.Command()
+
+	getLog, err := document.Get(command)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if _, ok := getLog.(types.NullType); ok {
+		return nil, commonerrors.NewCommandErrorMsg(
+			commonerrors.ErrMissingField,
+			`BSON field 'getLog.getLog' is missing but a required field`,
+		)
+	}
+
+	if _, ok := getLog.(string); !ok {
+		return nil, commonerrors.NewCommandError(
+			commonerrors.ErrTypeMismatch,
+			fmt.Errorf(
+				"BSON field 'getLog.getLog' is the wrong type '%s', expected type 'string'",
+				commonparams.AliasFromType(getLog),
+			),
+		)
+	}
+
+	var resDoc *types.Document
+
+	switch getLog {
+	case "*":
+		resDoc = must.NotFail(types.NewDocument(
+			"names", must.NotFail(types.NewArray("global", "startupWarnings")),
+			"ok", float64(1),
+		))
+
+	case "global":
+		log, err := logging.RecentEntries.GetArray(zap.DebugLevel)
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+		resDoc = must.NotFail(types.NewDocument(
+			"log", log,
+			"totalLinesWritten", int64(log.Len()),
+			"ok", float64(1),
+		))
+
+	case "startupWarnings":
+		state := h.StateProvider.Get()
+
+		info := version.Get()
+
+		startupWarnings := []string{
+			"Powered by FerretDB " + info.Version + " and SQLite.",
```

Comment:
Currently, we don't have handler version for SQLite, so I didn't include it.

Reply:
https://github.com/FerretDB/FerretDB/blob/b16fe1f85de627e6cdcee186b21e56e2f13e6974/internal/backends/sqlite/metadata/pool/pool_test.go#L169

Reply:
Great! Thanks!